Lickability/PinpointKit
# Swift 2.3 Maintenance

## What it Does

Maintenance work on the `swift-2.3` branch for Xcode 8.2.

* Addresses compiler warnings in `ScreenshotCell.swift`.
* Adds a [`.swift-version`](http://blog.cocoapods.org/CocoaPods-1.1.0/) file to specify `2.3` on this branch. The default is `3.0` when left unspecified.
* Accepts Xcode’s recommended changes for both the example project and the framework itself:

<img width="615" alt="screen shot 2017-01-13 at 10 54 25 am" src="https://cloud.githubusercontent.com/assets/7883805/21938240/7fc41674-d988-11e6-828f-33ac03739487.png">
<img width="615" alt="screen shot 2017-01-13 at 10 59 34 am" src="https://cloud.githubusercontent.com/assets/7883805/21938239/7fc3c048-d988-11e6-9f80-7a309e33e478.png">

## How to Test

1. Create a new Xcode project (Single View Application)
1. Change the build setting "Use Legacy Swift Language Version" to "Yes"
1. Run `pod init`
1. Run `open Podfile` and add `pod 'PinpointKit', :git => 'https://github.com/Lickability/PinpointKit.git', :branch => 'swift-2.3-maintenance'` to your target.
1. Run `pod install`
1. Close the project and open the newly created workspace
1. Choose "Later" when prompted to convert to current Swift syntax
1. Build (without error :tada:)
1. Open `ViewController.swift` and replace the contents with the snippet at the end of this series of steps.
1. Build and Run, verifying that PinpointKit’s UI appears

```swift
import UIKit
import PinpointKit

class ViewController: UIViewController {
    
    let pinpointKit = PinpointKit(feedbackRecipients: ["something@example.com"])

    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(true)
        
        pinpointKit.show(fromViewController: self)
    }
}
``` 

## Notes

This fixes the issue which results in compiler errors and the need to change the "Use Legacy Swift Language Version" to "Yes" in the PinpointKit target. This will still cause the following dialog to appear:

<img width="454" alt="screen shot 2017-01-13 at 11 52 13 am" src="https://cloud.githubusercontent.com/assets/7883805/21938665/43f7ec2c-d98a-11e6-8a1e-bd8c9ff6d274.png">

I believe this will be fixed by https://github.com/CocoaPods/CocoaPods/issues/6067#issuecomment-255516718 (in the next version). Currently "Use Legacy Swift Language Version" is left "Unspecified".
